presto-hive/src/test/java/com/facebook/presto/hive/metastore/MockHiveMetastoreClient.java

```diff
@@ -277,7 +278,25 @@ public PrincipalPrivilegeSet getPrivilegeSet(HiveObjectRef hiveObject, String us
     @Override
     public List<String> getRoleNames()
     {
-        throw new UnsupportedOperationException();
+        accessCount.incrementAndGet();
+        if (throwException) {
+            throw new IllegalStateException();
+        }
+        return TEST_ROLES;
+    }
+
+    @Override
+    public void createRole(String role, String grantor)
+            throws TException
+    {
+        // No-op
```

Comment:
Why not mimic e.g. `createTable`'s behavior - to throw the `UnsupportedOperationException`?